src/Behat/Behat/Output/Node/Printer/JUnit/JUnitSuitePrinter.php

```diff
@@ -44,7 +44,7 @@ public function printHeader(Formatter $formatter, Suite $suite)
 
         /** @var JUnitOutputPrinter $outputPrinter */
         $outputPrinter = $formatter->getOutputPrinter();
-        $outputPrinter->createNewFile(!empty($formatter->fileName) ? $formatter->fileName : $suite->getName());
+        $outputPrinter->createNewFile(!empty($name = $formatter->getName()) ? $name : $suite->getName());
```

Comment:
1. You are asking `$formatter` to give you the name
2. You are also asking `$suite` to give you the name
3. You then create a file name on the spot from these two pieces of information

You basically just asked a person to hand you over their phone just to check the current time. Instead of simply asking the person what time is it. Imagine next time person forgets his phone at his desk, but still does wear a watch on his hand - your time-asking routine will completely fail, even though person *can* technically tell you what time is it. If only you asked...

You have `$formatter`, you have `$suite`, you have a behaviour of figuring out the filename based on the information these two have. Give that behaviour a name (e.g. `generateFilename()`) and assign it to one of these two objects (e.g. `$formatter->generateFilenameFor($suite)`).

Tell objects to tell you the time, don't ask for their phone to check the time...